task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -1147,8 +1151,22 @@ def final_state(self):
             return self._terminal_state or TaskInstanceState.SUCCESS
         if self._exit_code != 0 and self._terminal_state == SERVER_TERMINATED:
             return SERVER_TERMINATED
+
+        if self._exit_code != 0 and self._is_signal_retryable() and self._should_retry:
+            return TaskInstanceState.UP_FOR_RETRY
```

Comment:
This feels a little bit odd to me here.

My first thought was that we should do this inside the place where we first notice the signal exit code i.e. inside `_check_subprocess_exit` But I'm not sure.